pygmt/src/coast.py

```diff
@@ -183,10 +188,23 @@ def coast(self, **kwargs):
     {V}
     """
     kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
+
+    if "clip" in kwargs:
+        if kwargs["clip"] == "land":
+            kwargs["G"] = True
+            kwargs.pop("clip")
+        elif kwargs["clip"] == "water":
+            kwargs["S"] = True
+            kwargs.pop("clip")
+        elif kwargs["clip"] == "end":
+            kwargs["Q"] = True
+            kwargs.pop("clip")
```

Comment:
Check if the given clip parameter is valid.
```suggestion
    if "clip" in kwargs:
        if kwargs["clip"] in ["land", "water", "end"]:
            kwargs.pop("clip")
            if kwargs["clip"] == "land":
                kwargs["G"] = True
            elif kwargs["clip"] == "water":
                kwargs["S"] = True
            elif kwargs["clip"] == "end":
                kwargs["Q"] = True
        else:
            raise GMTInvalidInput(
                "Invalid clip parameter. Must be one of 'land', 'water', and 'end'."
            )  
```